src/main/java/seedu/address/model/reminder/UniqueReminderList.java

```diff
@@ -112,14 +107,30 @@ public ObservableList<Reminder> getRemindersAfterDate(Date date) {
     }
 
     /**
-     * Resets the internal list of reminders using the given list of persons.
+     * Updates the entire internal list of reminders using the given list of persons.
      *
      * @param personList The persons from which the reminders are to be produced.
      */
-    public void setReminders(ObservableList<Person> personList) {
+    public void updateReminders(ObservableList<Person> personList) {
+        requireNonNull(personList);
+        internalList.clear();
         for (Person person : personList) {
-            this.add(person);
+            person.updateReminder();
+            person.getReminder().ifPresent(internalList::add);
+        }
+    }
+
+    /**
+     * Updates the internal list of reminders if the list is dirty.
+     * @param personList
+     */
+    public void updateRemindersIfDirty(ObservableList<Person> personList) {
+        if (!isReminderListDirty) {
+            return;
         }
+
+        updateReminders(personList);
+        isReminderListDirty = false;
```

Comment:
Shouldn't one reminder list be associated with one PersonList via Model, hence these two methods don't really need `personList`? This current implementation made the following scenario possible:
I set the reminderlist as clean to update a different person list created somewhere along the line of this code, even though the reminderList which should represent the personList I am concerned with is still dirty (or just gone untraced in this case).

Reply:
Oh really good point

Reply:
I'm not sure how to fix it though.

The only way to get personlist from model is with getFilteredPersonsList. Which means it could be that when I pull up the dashboard (and update the updated persons list lazily) wrong people get updated

Reply:
WAIT NEVERMIND!
I will just used model.getAddressBook().getPersonList()